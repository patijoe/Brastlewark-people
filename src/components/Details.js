import React from 'react';
import { Link } from 'react-router-dom';

class Details extends React.Component {

  componentWillUnmount() {
    this.props.handleResetFilter();
  }

  render() {
    const { population } = this.props;
    const id = this.props.match.params.id;

    const newPerson = population.find(item => parseInt(id) === item.id);

    return (
      <div className="details__page">
        {newPerson !== undefined
          ? <div className="detail__container">

            <div className="newPerson__img" style={{ backgroundImage: `url(${newPerson.thumbnail})` }}></div>

            <div className="newPerson__info">
              <h2 className="newPerson__name">{newPerson.name}</h2>

              <div className="newPerson__block1">

                <div className="newPerson__char">
                  <h2 className="newPerson__details">Personal characteristics</h2>
                  <h3 className="newPerson__age char">Age: {newPerson.age}</h3>
                  <h3 className="newPerson__weight char">Weight: {newPerson.weight}</h3>
                  <h3 className="newPerson__height char">Heigth: {newPerson.height}</h3>
                  <h3 className="newPerson__hair char">Color hair: {newPerson.hair_color}</h3>
                </div>

                <div className="newPerson__prof">
                  <h2 className="newPerson__details">Professions</h2>
                  <ul className="prof__list">
                    {newPerson.professions.map(item => {
                      return (<li className="prof__item char">{item}</li>);
                    })}
                  </ul>
                </div>
              </div>

              <h2 className="newPerson__details">Friendships</h2>
              <h3 className="newPerson__friends char">
                {newPerson.friends.join(', ')}
              </h3>
            </div>

          </div>
          : 'no hay info'}

        <Link className="return__link" to="/">RETURN</Link>
      </div>
    );
  }
}

export default Details;