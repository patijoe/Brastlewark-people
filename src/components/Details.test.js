import React from 'react';
import { create } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Details from './Details';

const population = [
  {
    thumbnail: 'https://estaticos.elperiodico.com/resources/jpg/1/3/1554908865931.jpg',
    name: 'lola',
    age: 14,
    weight: 50,
    height: 161,
    hair_color: 'brown',
    proffesions: ['fronEnd'],
    friends: ['mario']
  }
]

const match = {
  params: {
    id: 1 
  }
};

describe('Details component', () => {
  it('matches the snapshot in Details component', () => {
    const component = create (
      <Router>
        <Details match={match} population={population} />
      </Router>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});