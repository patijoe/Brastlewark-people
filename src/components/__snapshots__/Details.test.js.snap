// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component matches the snapshot in Details component 1`] = `
<div
  className="details__page"
>
  no hay info
  <a
    className="return__link"
    href="/"
    onClick={[Function]}
  >
    RETURN
  </a>
</div>
`;
